targets/l2_switch/CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+# Include directories
+include_directories(
+  ${CMAKE_CURRENT_SOURCE_DIR}
+  ${CMAKE_CURRENT_BINARY_DIR}
+)
+
+# Create l2switch library
+add_library(l2switch STATIC
+  l2_switch.cpp
+  learn_client.cpp
+)
+
+target_link_libraries(l2switch
+  bmsim
+  bflpmtrie
+  bmi
+  jsoncpp
+  Boost::system
+  Boost::program_options
+  Boost::filesystem
+  ${THRIFT_LIBRARIES}
+)
+
+# Create l2_switch executable
+add_executable(l2_switch
```

Comment:
Are you hitting any issue having a library and executable with the same name? A Google search indicates that this used to be a problem -- but perhaps its not any more?

This was the recommended solution for having a library and executable with the same name: https://discourse.cmake.org/t/how-to-use-the-same-name-for-a-target-and-a-library/5044